File: hw2/CSE465-565/Chapter6DataTypes/Overflow.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OverflowCS {
	public class OverflowExample {
		public static void Display(double v) {
			if (Double.IsInfinity(v)) {
				Console.WriteLine("INFINITY");
			} else if (Double.IsNaN(v)) {
				Console.WriteLine("NAN");
			} else {
				Console.WriteLine(v);
			}
		}
		public static void Main(String [] args) {
			int x = 1;
			int cnt = 0;
			Display(1.4/0.0);
			Display(0.0/0.0);
			Display(Math.Sqrt(-1.0));
			Display(Double.NegativeInfinity / Double.NegativeInfinity );
			Display(Double.PositiveInfinity  / Double.NegativeInfinity );
			Display(Double.NegativeInfinity  / Double.PositiveInfinity);
			Display(Double.PositiveInfinity / Double.PositiveInfinity);

			x = 1;
			while (x > 0) {
				cnt++;
				x *= 2;
			}
			Console.WriteLine(cnt + " " + x);

			x = 1;
			try {
				while (x > 0) {
					cnt++;
					x = checked(x * 2);
				}
				Console.WriteLine(cnt + " " + x);
			}
			catch (System.OverflowException e) {
				Console.WriteLine(e.StackTrace);
			}
		}
	}
}

File: hw2/CSE465-565/Chapter6DataTypes/overflow.java
import java.math.*;

public class overflow {
	public static void Display(double v) {
		if (Double.isInfinite(v)) {
			System.out.println("INFINITY");
		} else if (Double.isNaN(v)) {
			System.out.println("NAN");
		} else {
			System.out.println(v);
		}
	}
	public static void main(String [] args) {
		int x = 1;
		int cnt = 0;
		Display(1.4/0.0);
		Display(0.0/0.0);
		Display(Math.sqrt(-1.0));
		Display(Double.NEGATIVE_INFINITY / Double.NEGATIVE_INFINITY);
		Display(Double.POSITIVE_INFINITY / Double.NEGATIVE_INFINITY);
		Display(Double.NEGATIVE_INFINITY / Double.POSITIVE_INFINITY);
		Display(Double.POSITIVE_INFINITY / Double.POSITIVE_INFINITY);

		while (x > 0) {
			cnt++;
			x *= 2;
		}

		System.out.println(cnt + " " + x);
	}
}
